cmd/main.go

```diff
@@ -202,9 +191,22 @@ func newCachedClient(ctx context.Context, cfg *rest.Config) (client.Client, erro
 
 	// populate the cache backed by shared informers that are initialized lazily on the first call
 	// for the given GVK with all resources we are interested in from the host-operator namespace
-	objectsToList := []client.ObjectList{&toolchainv1alpha1.ToolchainConfigList{}, &corev1.SecretList{}}
-	for i := range objectsToList {
-		if err := hostCluster.GetClient().List(ctx, objectsToList[i], client.InNamespace(configuration.Namespace())); err != nil {
+	objectsToList := map[string]client.ObjectList{
+		"MasterUserRecord": &toolchainv1alpha1.MasterUserRecordList{},
+		"Space":            &toolchainv1alpha1.SpaceList{},
+		"SpaceBinding":     &toolchainv1alpha1.SpaceBindingList{},
+		"ToolchainStatus":  &toolchainv1alpha1.ToolchainStatusList{},
+		"UserSignup":       &toolchainv1alpha1.UserSignupList{},
+		"ProxyPlugin":      &toolchainv1alpha1.ProxyPluginList{},
+		"NSTemplateTier":   &toolchainv1alpha1.NSTemplateTierList{},
+		"ToolchainConfig":  &toolchainv1alpha1.ToolchainConfigList{},
+		"BannedUser":       &toolchainv1alpha1.BannedUserList{},
+		"Secret":           &corev1.SecretList{}}
```

Comment:
I believe Matous added these resources based on the informers we were creating in informers.go. If my memory serves me correctly, the list of resources there are those that are needed in the proxy flow and so we pre-populate the cache for those resources so that the proxy is as fast as possible even on first time use by a user. I don't remember whether ConfigMaps are part of that flow but I don't think it hurts to add it.

Reply:
@rajivnathan is correct, we want to cache only those resources we access in reg-service. Caching anything else doesn't make much sense. 

We could theoretically drop a list of some reources like Secrets, NSTemplateTiers, ProxyPlugins, ToolchainStatuses, and ToolchainConfigs because we don't expect many resources to be present in the namespace, so the very first call wouldn't take much more time compared to when it is already cached. But I thought that it's also completely fine to pre-populate the cache with all resources the reg-service and proxy touch.

Reply:
Cache can be crucial for performance. Especially for the proxy. And it's easy to miss this (don't add specific resources to the initialization step) during development. We don't have any performance tests in our CI.
If neither reg-service or proxy touches CMs then IMO we should remove it from the role.
And I would argue that it would be safer to keep everything reg-service & proxy access in the cache initialization.

I checked it looks like the CM are not used by reg-service and proxy. So let's create a separate PR to remove it from the role.